src/Orleans.Transactions/DistributedTM/TransactionAgent.cs

```diff
@@ -93,6 +93,9 @@ public Task<ITransactionInfo> StartTransaction(bool readOnly, TimeSpan timeout)
                         if (logger.IsEnabled(LogLevel.Debug))
                             logger.Debug($"{stopwatch.Elapsed.TotalMilliseconds:f2} fail {transactionInfo.TransactionId} prepare response status={status}");
 
+                        foreach (var p in participants)
+                            p.Key.Abort(transactionInfo.TransactionId).Ignore();
```

Comment:
Shouldn't we wait for the aborts?  If not, won't this allow future calls to fail because they may depend on the aborted operation.